From cdc4bc6fa37e3d5173b3cad3d0f2e2d9816f8eb1 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Marek=20Marczykowski-G=C3=B3recki?=
 <marmarek@invisiblethingslab.com>
Date: Sun, 15 Dec 2024 16:13:42 +0100
Subject: [PATCH 1/2] Drop versioned dependencies

Upstream bumps version requirements arbitrarily, in most cases without
any specific API or functional requirement.
Use package version available in the distribution instead.

And also cryptography was twice.
---
 requirements/base.txt   | 12 ++++++------
 requirements/zeromq.txt |  3 +--
 2 files changed, 7 insertions(+), 8 deletions(-)

diff --git a/requirements/base.txt b/requirements/base.txt
index 0061087e35..a42ff50f7d 100644
--- a/requirements/base.txt
+++ b/requirements/base.txt
@@ -7,29 +7,27 @@
 PyYAML
 MarkupSafe
-requests<2.32.0 ; python_version < '3.10'
-requests>=2.32.3 ; python_version >= '3.10'
+requests
-certifi==2023.07.22; python_version < '3.10'
-certifi>=2024.7.4; python_version >= '3.10'
+certifi
 distro>=1.0.1
 psutil<6.0.0; python_version <= '3.9'
 psutil>=5.0.0; python_version >= '3.10'
 # Packaging 24.1 imports annotations from __future__ which breaks salt ssh
 # tests on target hosts with older python versions.
-packaging==24.0
+packaging>=21.3
 looseversion
-tornado>=6.4.1
-aiohttp>=3.9.0
+tornado
+aiohttp

 croniter>=0.3.0,!=0.3.22; sys_platform != 'win32'

 setproctitle>=1.2.3
-timelib>=0.2.5
-pyopenssl>=24.0.0
+timelib
+pyopenssl
 python-dateutil>=2.8.1
 python-gnupg>=0.4.7
 cherrypy>=18.6.1
 importlib-metadata>=3.3.0
-cryptography>=42.0.0
+cryptography

 # From old windows.txt requirements file
 gitpython>=3.1.37; sys_platform == 'win32'
@@ -44,10 +43,7 @@
 
 # We need contextvars for salt-ssh

-cryptography>=42.0.0
+urllib3

-urllib3>=1.26.20,<2.0.0; python_version < '3.10'
-urllib3>=2.5.0; python_version >= '3.10'
-
-jaraco.text>=4.0.0
-jaraco.functools>=4.1.0
+jaraco.text
+jaraco.functools
diff --git a/requirements/zeromq.txt b/requirements/zeromq.txt
index 4000f5eb01..ec03558d33 100644
--- a/requirements/zeromq.txt
+++ b/requirements/zeromq.txt
@@ -1,4 +1,4 @@
 -r base.txt
 -r crypto.txt
 
-pyzmq>=25.1.2
+pyzmq
-- 
2.46.0
